src/coreclr/vm/callstubgenerator.cpp

```diff
@@ -1975,12 +1975,19 @@ extern "C" void InterpreterStubRetVoid();
 extern "C" void InterpreterStubRetDouble();
 extern "C" void InterpreterStubRetI8();
 
-#if defined(TARGET_WINDOWS) && defined(TARGET_AMD64)
+#ifdef TARGET_AMD64
+#ifdef TARGET_WINDOWS
 extern "C" void CallJittedMethodRetBuffRCX(PCODE *routines, int8_t*pArgs, int8_t*pRet, int totalStackSize);
 extern "C" void CallJittedMethodRetBuffRDX(PCODE *routines, int8_t*pArgs, int8_t*pRet, int totalStackSize);
 extern "C" void InterpreterStubRetBuffRCX();
 extern "C" void InterpreterStubRetBuffRDX();
-#else // TARGET_WINDOWS && TARGET_AMD64
+#else // TARGET_WINDOWS
+extern "C" void CallJittedMethodRetBuffRDI(PCODE *routines, int8_t*pArgs, int8_t*pRet, int totalStackSize);
+extern "C" void CallJittedMethodRetBuffRSI(PCODE *routines, int8_t*pArgs, int8_t*pRet, int totalStackSize);
+extern "C" void InterpreterStubRetBuffRDI();
+extern "C" void InterpreterStubRetBuffRSI();
+#endif // TARGET_WINDOWS
```

Comment:
Can we instead abstract this away in the implementation rather than putting it in the name of the function? I'm assuming we are going for `RetBufFirstArg` and `RetBufSecondArg`. I find the approach we take where we encode the register name in the method name to make consuming code cumbersome and encourage the proliferation of defines.

Reply:
Okay. I'd like to revisit this at a later date though.